src/xnode_admin/xnode_builder.py

```diff
@@ -338,33 +361,22 @@ def process_studio_config(studio_json_config, state_directory):
         f.write(new_sys_config)
 
 
-def os_channel(update_or_rollback: bool):
+def flake_update(state_directory):
     # Update the channel.
-    argument = ""
-    if update_or_rollback:
-        argument = "--update"
-    else:
-        argument = "--rollback"
-
-    result = subprocess.run(['/run/current-system/sw/bin/nix-channel', argument], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    result = subprocess.run(['/run/current-system/sw/bin/nix', 'flake', 'update', '--flake', state_directory], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
You might have to add '--extra-experimental-features 'command flakes'` to this command if you see an error about the missing feature